src/SprykerSdk/Acp/Configuration/SchemaGenerator/SchemaGenerator.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Acp\Configuration\SchemaGenerator;
+
+use cebe\openapi\spec\OpenApi;
+use SprykerSdk\Acp\Configuration\Reader\AppConfigurationReaderInterface;
+
+class SchemaGenerator implements SchemaGeneratorInterface
+{
+    protected AppConfigurationReaderInterface $appConfigurationReader;
+
+    protected SchemaWriterInterface $schemaWriter;
+
+    /**
+     * @param \SprykerSdk\Acp\Configuration\Reader\AppConfigurationReaderInterface $appConfigurationReader
+     * @param \SprykerSdk\Acp\Configuration\SchemaGenerator\SchemaWriterInterface $schemaWriter
+     */
+    public function __construct(AppConfigurationReaderInterface $appConfigurationReader, SchemaWriterInterface $schemaWriter)
+    {
+        $this->appConfigurationReader = $appConfigurationReader;
+        $this->schemaWriter = $schemaWriter;
+    }
+
+    /**
+     * @param string $configurationFilePath
+     * @param string $openapiFilePath
+     *
+     * @return bool
+     */
+    public function convertConfigurationToSchema(string $configurationFilePath, string $openapiFilePath): bool
```

Comment:
Since this method does conversion and writing to the underlying OpenAPI file, would `addConfigurationToSchema` be a better name?

Reply:
Alternatively, should we move the responsibility of reading/writing files to the client, so this class is only concerned about converting data structures? This seems more in line with SRP.